frontend/packages/ux-editor/src/containers/DesignView/PageAccordion/PageAccordion.tsx

```diff
@@ -52,16 +52,33 @@ export const PageAccordion = ({
 
   const deleteLayout = useDeleteLayout();
 
+  const [isDeletionInProgress, setIsDeletionInProgress] = useState(false);
   const handleConfirmDelete = useCallback(() => {
-    if (confirm(t('ux_editor.page_delete_text'))) {
-      deleteLayout(pageName);
-
-      if (selectedLayout === pageName) {
-        const layoutToSelect = firstAvailableLayout(pageName, layoutOrder);
-        setSearchParams({ layout: layoutToSelect });
-      }
+    if (!isDeletionInProgress) {
+      setIsDeletionInProgress(true);
     }
-  }, [deleteLayout, layoutOrder, pageName, selectedLayout, setSearchParams, t]);
+  }, [isDeletionInProgress]);
+  useEffect(() => {
+    const executeDeletion = async () => {
+      if (isDeletionInProgress && confirm(t('ux_editor.page_delete_text'))) {
+        deleteLayout(pageName);
+        if (selectedLayout === pageName) {
+          const layoutToSelect = firstAvailableLayout(pageName, layoutOrder);
+          setSearchParams({ layout: layoutToSelect });
+        }
+      }
+      setIsDeletionInProgress(false);
+    };
+    executeDeletion();
+  }, [
+    isDeletionInProgress,
+    deleteLayout,
+    layoutOrder,
+    pageName,
+    selectedLayout,
+    setSearchParams,
+    t,
+  ]);
```

Comment:
`deleteLayout` er ikke asynkron, så dette vil vel ikke gjøre noen forskjell? `isDeletionInProgress` blir bare satt tilbake til `false` når komponenten blir lastet på nytt, så den vil ikke kunne si noe sikkert om hvorvidt slettingen faktisk pågår. Vi burde heller bruke de innebygde verktøyene til React Query her. I `useDeleteLayout` kan du hente ut `isPending` fra `useDeleteLayoutMutation` og gjøre den tilgjengelig for `PageAccordion`. Da vil komponenten automatisk lastes på nytt når kallet er fullført med riktig verdi i `isPending`.